providers/base/units/firmware/jobs.pxu

```diff
@@ -105,3 +105,14 @@ plugin: attachment
 depends: firmware/fwts_dump
 command:
   [ -f "$PLAINBOX_SESSION_SHARE/acpidump.log" ] && gzip -c "$PLAINBOX_SESSION_SHARE/acpidump.log"
+
+id: firmware/fwupdmgr_get_devices
+plugin: attachment
+category_id: com.canonical.plainbox::firmware
+_summary: Collect the device firmware update information
+_purpose: Attach information about the devices, as reported by the fwupdmgr
+requires:
+    executable.name == "fwupdmgr"
+command:
+    unset SNAP
```

Comment:
If you upgrade fwupd to 1.9.14/15 in 22.04 you won't need to do this.

Reply:
@superm1  
thanks for the feedback.
But on the PC enablement project, system will include the fwupd Debian package instead of fwupd snap package.
And considering the fwupd Debian package is installed by default, so we should use it for the testing.
So we still need to apply the workaround for now, until we get any fixed for fwupd Debian package.

Reply:
Right; I just mean to say you can do backport/SRU and it can be dropped or you can have different test case for different releases.

Reply:
I see your concerns now, but this test case would be included across different Ubuntu LTS version.
So I have implemented a small scripts to detect the fwupd package to allow check could retrieve firmware information  with Debian fwupd and Snap fwupd now.

Reply:
@superm1  let me know if you have any concerns with current implementation.

Reply:
No it looks fine to me. The only suggestion I would offer you is that you can totally avoid a lot of this if you just use gobject introspection instead. Then you don't need to distinguish any versions snap or deb. You would just use the libfwupd gi instead.

You can look at simple_client.py upstream to see what I'm talking about.

But if you choose not to do it what you did looks functional.

Reply:
I think you must be missing a slot or connection to properly communicate with confined snap. But yeah that's fine.